Source/ARTEncoder.h

```diff
@@ -43,8 +43,8 @@ ART_ASSUME_NONNULL_BEGIN
 - (art_nullable NSData *)encodePresenceMessage:(ARTPresenceMessage *)message;
 - (art_nullable NSData *)encodePresenceMessages:(NSArray *)messages;
 
-- (art_nullable NSData *)encodeProtocolMessage:(ARTProtocolMessage *)message;
-- (art_nullable ARTProtocolMessage *)decodeProtocolMessage:(NSData *)data;
+- (art_nullable NSData *)encodeProtocolMessage:(ARTProtocolMessage *)message error:(NSError * __autoreleasing *)error;
```

Comment:
I did that because I took a look at the `NSJSONSerialization` header file that returns errors as well 😅 `[NSJSONSerialization JSONObjectWithData:<#(nonnull NSData *)#> options:<#(NSJSONReadingOptions)#> error:<#(NSError * _Nullable __autoreleasing * _Nullable)#>]`